### PR TITLE
[Python] Added load model option args for PyTorch Model Handler

### PR DESCRIPTION
**Pull request by riteshghorse**
Added optional args to PyTorch model handler in case of custom config for saving/loading the model.
https://pytorch.org/docs/stable/generated/torch.load.html

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by riteshghorse** [comment]
This test is failing  `FAILED apache_beam/transforms/ptransform_test.py::DeadLettersTest_0::test_timeout` unrelated to this PR.

R: @AnandInguva

File: sdks/python/apache_beam/ml/inference/pytorch_inference_test.py [hunk 1]
```diff
@@ -893,5 +894,55 @@ def test_env_vars_set_correctly_keyed_tensor_handler(self):
...
+    load_model_args = {'weights_only': True}
+    model_handler = PytorchModelHandlerTensor(
+        state_dict_path=self.torch_path,
+        model_class=PytorchLinearRegression,
+        model_params=self.model_params,
+        load_model_args=load_model_args)
+    model_handler.load_model()
+    pytorch_inference._load_model.assert_called_with(
+        PytorchLinearRegression,
+        self.torch_path,
```
Comment (AnandInguva) [inline sdks/python/apache_beam/ml/inference/pytorch_inference_test.py:924]: Can we pass the args using keywords? just for readability purpose

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 2]
```diff
@@ -222,6 +225,8 @@ def __init__(
         batch will be fed into the inference_fn as a Sequence of Tensors.
       max_batch_size: the maximum batch size to use when batching inputs. This
         batch will be fed into the inference_fn as a Sequence of Tensors.
+      load_model_args:a dictionary of parameters to pass to the torch.load
```
Comment (AnandInguva) [inline sdks/python/apache_beam/ml/inference/pytorch_inference.py:228]:
```suggestion
      load_model_args: a dictionary of parameters passed to the torch.load
```

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 3]
```diff
@@ -441,6 +449,8 @@ def __init__(
         batch will be fed into the inference_fn as a Sequence of Keyed Tensors.
       max_batch_size: the maximum batch size to use when batching inputs. This
         batch will be fed into the inference_fn as a Sequence of Keyed Tensors.
+      load_model_args:a dictionary of parameters to pass to the torch.load
```
Comment (AnandInguva) [inline sdks/python/apache_beam/ml/inference/pytorch_inference.py:452]:
```suggestion
      load_model_args: a dictionary of parameters passed to the torch.load
```

File: sdks/python/apache_beam/ml/inference/pytorch_inference.py [hunk 4]
```diff
@@ -222,6 +225,8 @@ def __init__(
         batch will be fed into the inference_fn as a Sequence of Tensors.
       max_batch_size: the maximum batch size to use when batching inputs. This
         batch will be fed into the inference_fn as a Sequence of Tensors.
+      load_model_args: a dictionary of parameters passed to the torch.load
+          function of to specify custom config for loading models.
```
Comment (AnandInguva) [inline sdks/python/apache_beam/ml/inference/pytorch_inference.py:229]: nit: 
```suggestion
          function to specify custom config for loading models.
```

Same below

**Review by AnandInguva — APPROVED** [review]

**Comment by riteshghorse** [comment]
retest this please

**Pull request merged**